### PR TITLE
Add haproxy

### PR DESCRIPTION
**Pull request by mr-sandy**

### CODE REVIEW

**Review by bazwilliams — APPROVED** [review]
LFTM

**Comment by bazwilliams** [comment]
I presume you'll also push to the proxies repo?

**Comment by wilhen01** [comment]
Woop Woop! So this uses a Windows environment variable called COMPUTERNAME? Is that in fact a built in (i.e. pre-defined) environment variable?

**Comment by bazwilliams** [comment]
Yep, so this should be backwards compatible with you cygwin fans!

**Comment by wilhen01** [comment]
Noone is a fan of cygwin 😛 

I can't wait to move to docker. Will wait til Colin's built my new machine though.

File: docker-compose.yml [hunk 1]
```diff
@@ -27,3 +27,15 @@ services:
           - "27017:27017"
       volumes:
           - /var/lib/mongodb
+
+    haproxy:
+      image: haproxy:1.7
+      ports:
+          - "80:80"
+      volumes:
+          - /c/projects/proxies/localhost/etc/haproxy:/usr/local/etc/haproxy/
```
Comment (liddellj) [inline docker-compose.yml]: Does everyone store their projects under this path? Otherwise this won't work...

Comment (richardiphillips) [reply docker-compose.yml]: I don't. I'm the lightning rod that hits the failure every time people hard-code this path and it's happened quite a few times.

Comment (wilhen01) [reply docker-compose.yml]: @richardiphillips There's a simple solution to that problem 😉

Comment (bazwilliams) [reply docker-compose.yml]: Maybe we could use git submodule?

Comment (richardiphillips) [reply docker-compose.yml]: @wilhen01 Sorry. I can't risk the damage to our code base if I stop stemming the tide of hard-coded paths.

**Pull request merged**